src/wasm/activity-service/activity/fit/service.go

```diff
@@ -137,36 +161,38 @@ func (s *service) convertToActivity(activityFile *filedef.Activity) activity.Act
 			laps = append(laps, lap)
 		}
 
-		ses := activity.NewSessionFromLaps(laps, typedef.SportGeneric)
+		ses := activity.NewSessionFromLaps(laps)
 		ses.Records = records
 		ses.Laps = laps
-		s.preprocessingRecords(ses.Records, ses.Sport)
 		act.Sessions = []activity.Session{ses}
 
 		return act
 	} else if len(laps) == 0 {
-		// Some devices may only create sessions without laps, to ensure we don't lose any summary data in session
-		// since there are information that only available in session but not in records.
+		// Some devices may only create sessions without laps.
 		// Let's create laps from sessions, 1 session should at least have 1 lap.
 		laps = make([]activity.Lap, len(sessions))
 		for i := range sessions {
 			laps[i] = activity.NewLapFromSession(&sessions[i])
 		}
 	}
 
+	const anomalyThreshold = 10
```

Comment:
btw, what the use of this const?

Reply:
FIT files generated by Strava has 1 anomaly record that does not belong to any session due to the record timestamp is outside session's time-range. Moreover, FIT files may be truncated, leaving few records without session, instead of creating session for these few records, it's better to include it to previous session. Why 10? actually it's just an arbitrary number, having 10 or less records in a session do not significantly add any value.

Reply:
Typically a record is sampled per 1 second rate, so 10 records is usually worth 10 seconds. Unless the athlete is a sprinter, it does not worth much. Even if the user of our app is indeed a professional athlete, I believe professional athlete should have properly encoded FIT files without the anomaly.